### PR TITLE
#162948995 Users should be able to rate articles 

### PR DESCRIPTION
**Pull request by eduhmik**
### What does this PR do?
It allows users who are authenticated into our application to rate articles from 1-5

### Description of the PR?
- Allow users logged in the system to view and rate articles
- Restrict owners of articles from rating their own articles
- Get all ratings and return an average of the article ratings
- Return number of users who rated the article and a list of the ratings

### How can this PR be manually tested?
Run the server locally using `python manage.py runserver`
Try this endpoints, the method is a `POST`;
`http://127.0.0.1:8000/api/v1/articles/{{slug}}/rate/` by sending this payload
```
{	
	"rating":{
		"rating": 5
	}
}
```
This will give a rating to your specific article who slug is appended in the url

View all the ratings of a specific article using the endpoint below, the method is a `GET`;
`http://127.0.0.1:8000/api/v1/articles/{{slug}}/ratings/`

### Any background context about this PR?
None

### What are the relevant Pivotal Tracker stories?
[162948995](https://www.pivotaltracker.com/story/show/162948995)

## Rate an article
screenshot 2019-01-24 at 17 15 37

## View Ratings on our Article
screenshot 2019-01-24 at 17 16 07

## View Ratings of a specific Article
screenshot 2019-01-24 at 17 16 28

### CODE REVIEW

File: authors/apps/articles/serializers.py [hunk 1]
```diff
@@ -48,12 +52,28 @@ class Meta:
...
             'slug',
+            'avg_rating',
             'image_url',
             'author',
             'created_at',
             'updated_at'
         )
 
+    def get_average_rating(self, instance):
+        avg_rating = ArticleRatings.objects.filter(article=instance).aggregate(
```
Comment (charitymarani) [inline authors/apps/articles/serializers.py:63]: avg_rating is not descriptive enough. Consider using average_rating

File: authors/apps/articles/renderers.py [hunk 2]
```diff
@@ -24,3 +24,23 @@ def render(self, data, accepted_media_type=None, renderer_context=None):
...
+
+class RatingJSONRenderer(renderers.JSONRenderer):
+    charset = 'utf-8'
+
+    media_type = 'application/json'
+    format = 'json'
+    
+    def render(self, data, accepted_media_type=None, renderer_context=None):
+        if isinstance(data, dict):
+                # converts it into a dictionary with a key
```
Comment (charitymarani) [inline authors/apps/articles/renderers.py]: I don't think these comments are necessary

File: authors/apps/articles/serializers.py (hunk 1 above)
Comment (eduhmik) [reply authors/apps/articles/serializers.py:63]: Thanks @Charity... Fixing it

File: authors/apps/articles/renderers.py (hunk 2 above)
Comment (eduhmik) [reply authors/apps/articles/renderers.py]: Sure... I'm removing them

File: authors/apps/articles/views.py [hunk 3]
```diff
@@ -199,3 +202,67 @@ def update(self, request, slug, id):
...
+                    rating, data=serializer_data, partial=True)
+                serializer.is_valid(raise_exception=True)
+
+                serializer.save(rated_by=request.user, article=article)
+                data = serializer_data
+                data['message'] = "You have rated this article successfully"
+                return Response(data, status=status.HTTP_201_CREATED)
+
+
+class AvRatingsAPIView(RetrieveAPIView):
```
Comment (Nyakaru) [inline authors/apps/articles/views.py]: Maybe you should consider a better way of naming classes in the views.

File: authors/apps/articles/serializers.py (hunk 1 above)
Comment (eduhmik) [reply authors/apps/articles/serializers.py:63]: Fixed

**Pull request merged**